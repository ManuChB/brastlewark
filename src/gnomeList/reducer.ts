import {
    SAVE_DATA,
    SHOW_DETAILS,
    CLOSE_MODAL,
    SET_FILTERED_DATA
} from './types';
import { IGnomeListState } from './gnomeList';
import { Action } from 'redux';

const defaultState = {
    text: '',
    data: [],
    showModal: false,
    gnomeDetails: undefined,
    gnomeDetailsFriends: [],
    filteredData: []
};

const formReducer = (state: IGnomeListState = defaultState, action: {type: string, payload?: any}) => {
    switch (action.type) {
        case SAVE_DATA:
            return {
                ...state,
                data: action.payload
            };
        case SHOW_DETAILS:
            return {
                ...state,
                showModal: true,
                gnomeDetails: action.payload.gnome,
                gnomeDetailsFriends: action.payload.friends
            };
        case CLOSE_MODAL:
            return {
                ...state,
                showModal: false
            };

        case SET_FILTERED_DATA:
            return {
                ...state,
                filteredData: action.payload
            };
        default:
            return state;
    }
};

export default formReducer;