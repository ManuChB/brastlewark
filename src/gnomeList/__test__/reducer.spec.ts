import action from '../actions';
import reducer from '../reducer';

describe('GnomeList reducer', () => {
    const defaultState = {
        text: '',
        data: [],
        showModal: false,
        gnomeDetails: undefined,
        gnomeDetailsFriends: [],
        filteredData: []
    };

    const gnome = {
        id: 1,
        name: 'name',
        thumbnail: 'image',
        age: 100,
        weight: 100,
        height: 100,
        hair_color: 'red',
        professions: [],
        friends: [],
        onPress: jest.fn()
    }
    const gnomeList = [gnome]

    it('handle saveData()', () => {
        expect(reducer(defaultState, action.saveData(gnomeList))).toMatchSnapshot();
    });

    it('handle showGnomeDetails()', () => {
        expect(reducer(defaultState, action.showGnomeDetails(gnome, gnomeList))).toMatchSnapshot();
    });

    it('handle closeModal()', () => {
        expect(reducer(defaultState, action.closeModal())).toMatchSnapshot();
    });

    it('handle setFilteredData()', () => {
        const loading = false;
        expect(reducer(defaultState, action.setFilteredData(gnomeList))).toMatchSnapshot();
    });

});
