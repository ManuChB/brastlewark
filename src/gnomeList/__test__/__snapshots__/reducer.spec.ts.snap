// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomeList reducer handle closeModal() 1`] = `
Object {
  "data": Array [],
  "filteredData": Array [],
  "gnomeDetails": undefined,
  "gnomeDetailsFriends": Array [],
  "showModal": false,
  "text": "",
}
`;

exports[`GnomeList reducer handle saveData() 1`] = `
Object {
  "data": Array [
    Object {
      "age": 100,
      "friends": Array [],
      "hair_color": "red",
      "height": 100,
      "id": 1,
      "name": "name",
      "onPress": [MockFunction],
      "professions": Array [],
      "thumbnail": "image",
      "weight": 100,
    },
  ],
  "filteredData": Array [],
  "gnomeDetails": undefined,
  "gnomeDetailsFriends": Array [],
  "showModal": false,
  "text": "",
}
`;

exports[`GnomeList reducer handle setFilteredData() 1`] = `
Object {
  "data": Array [],
  "filteredData": Array [
    Object {
      "age": 100,
      "friends": Array [],
      "hair_color": "red",
      "height": 100,
      "id": 1,
      "name": "name",
      "onPress": [MockFunction],
      "professions": Array [],
      "thumbnail": "image",
      "weight": 100,
    },
  ],
  "gnomeDetails": undefined,
  "gnomeDetailsFriends": Array [],
  "showModal": false,
  "text": "",
}
`;

exports[`GnomeList reducer handle showGnomeDetails() 1`] = `
Object {
  "data": Array [],
  "filteredData": Array [],
  "gnomeDetails": Object {
    "age": 100,
    "friends": Array [],
    "hair_color": "red",
    "height": 100,
    "id": 1,
    "name": "name",
    "onPress": [MockFunction],
    "professions": Array [],
    "thumbnail": "image",
    "weight": 100,
  },
  "gnomeDetailsFriends": Array [
    Object {
      "age": 100,
      "friends": Array [],
      "hair_color": "red",
      "height": 100,
      "id": 1,
      "name": "name",
      "onPress": [MockFunction],
      "professions": Array [],
      "thumbnail": "image",
      "weight": 100,
    },
  ],
  "showModal": true,
  "text": "",
}
`;
